patina_dxe_core/src/image.rs

```diff
@@ -482,12 +522,132 @@ impl DxeCoreGlobalImageData {
+                dxe_core_hob.entry_point,
+            )
+        };
+
+        // SAFETY: The DXE Core HOB information is valid and points to a valid PE image. This ensures that the start
+        // address and length are accurate.
+        let dxe_core_image_buffer = unsafe {
+            from_raw_parts(
+                dxe_core_hob.alloc_descriptor.memory_base_address as *const u8,
+                dxe_core_hob.alloc_descriptor.memory_length as usize,
+            )
+        };
+
+        let pe_info = UefiPeInfo::parse(dxe_core_image_buffer).expect("Failed to parse PE info for DXE Core");
+
+        let private_image_data =
+            PrivateImageData::new_from_static_image(image_info, dxe_core_image_buffer, entry_point, &pe_info);
+
+        let handle = core_install_protocol_interface(
+            Some(protocol_db::DXE_CORE_HANDLE),
+            efi::protocols::loaded_image::PROTOCOL_GUID,
+            private_image_data.image_info.as_ref() as *const efi::protocols::loaded_image::Protocol as *mut c_void,
+        )
+        .unwrap_or_else(|err| panic!("Failed to install dxe core image handle: {err:?}"));
+
+        assert_eq!(handle, protocol_db::DXE_CORE_HANDLE);
+
+        initialize_debug_image_info_table(system_table);
+        core_new_debug_image_info_entry(
+            EfiDebugImageInfoNormal::EFI_DEBUG_IMAGE_INFO_TYPE_NORMAL,
+            private_image_data.image_info.as_ref() as *const efi::protocols::loaded_image::Protocol,
+            handle,
+        );
+
+        self.private_image_data.insert(handle, private_image_data);
+    }
+
+    /// Validates that the provided parent handle is valid and has a loaded image protocol.
+    fn validate_parent(parent: efi::Handle) -> Result<(), EfiError> {
+        PROTOCOL_DB.validate_handle(parent).inspect_err(|err| log::error!("Invalid parent handle {err:?}"))?;
+
+        PROTOCOL_DB.get_interface_for_handle(parent, efi::protocols::loaded_image::PROTOCOL_GUID).map_err(|err| {
+            log::error!("Failed to get loaded image interface on the parent handle: {err:?}");
+            EfiError::InvalidParameter
+        })?;
+        Ok(())
+    }
+
+    /// Returns the image metadata by it's buffer using device path protocol.
+    fn locate_image_metadata_by_buffer(
+        image: &[u8],
+        file_path: *mut efi::protocols::device_path::Protocol,
+    ) -> (Vec<u8>, bool, *mut c_void, u32) {
+        if let Ok((_, device_handle)) = core_locate_device_path(efi::protocols::device_path::PROTOCOL_GUID, file_path) {
+            (image.to_vec(), false, device_handle, 0)
+        } else {
+            (image.to_vec(), false, protocol_db::INVALID_HANDLE, 0)
+        }
+    }
+
+    /// Returns the image metadata by it's file path using simple file system or load file protocols.
+    ///
+    /// Note: presently none of the supported methods return `from_fv` or `authentication_status`.
```

Comment:
Not sure if the `Note:` is still accurate. At least one of the branches is returning `true` for from_fv